packages/babel-types/scripts/generators/typescript.js

```diff
@@ -109,25 +116,113 @@ for (const type in t.NODE_FIELDS) {
   }
 }
 
-for (let i = 0; i < t.TYPES.length; i++) {
-  let decl = `export function is${t.TYPES[i]}(node: object | null | undefined, opts?: object | null): `;
-
-  if (t.NODE_FIELDS[t.TYPES[i]]) {
-    decl += `node is ${t.TYPES[i]};`;
-  } else if (t.FLIPPED_ALIAS_KEYS[t.TYPES[i]]) {
-    decl += `node is ${t.TYPES[i]};`;
-  } else {
-    decl += `boolean;`;
-  }
-
-  lines.push(decl);
+for (const typeName of t.TYPES) {
+  const result =
+    t.NODE_FIELDS[typeName] || t.FLIPPED_ALIAS_KEYS[typeName]
+      ? `node is ${typeName}`
+      : "boolean";
+
+  lines.push(
+    `export function is${typeName}(node: object | null | undefined, opts?: object | null): ${result};`,
+    // TypeScript 3.7: https://github.com/microsoft/TypeScript/pull/32695 will allow assert declarations
```

Comment:
We can use [typesVersions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html#version-selection-with-typesversions) to publish typings for typescript <3.7 and typescript >=3.7 at the same time, but we'll need to generate types twice.

I want to refactor the typescript generation to use .d.ts files as templates (to make it much easier to get editor feedback while modifying them) so I can try to do that at the same time.